WordPress/Classes/ViewRelated/Site Creation/FinalAssembly/SiteCreationRotatingMessageView.swift

```diff
@@ -0,0 +1,189 @@
+    private let statusStackView: UIStackView = {
+        let stackView = UIStackView()
+
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.alignment = .center
+        stackView.axis = .horizontal
+        stackView.distribution = .fillProportionally
+        stackView.alignment = .center
+        stackView.spacing = Margins.horizontalSpacing
+
+        return stackView
+    }()
+
+    // MARK: - UIView
+
+    /// Creates a new instance of the message view with the provided messages
+    /// - Parameter messages: An array of messages to display
+    /// - Parameter iconImage: The icon image to display before the status message
+    init(messages: [String], iconImage: UIImage) {
+        self.statusMessages = messages
+
+        self.statusImageView = UIImageView(image: iconImage)
+
+        super.init(frame: .zero)
+        configure()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Public Methods
+    func startAnimating() {
+        stopAnimating()
+
+        self.animationTimer = Timer.scheduledTimer(timeInterval: Parameters.totalStatusDisplayDuration,
+                                                   target: self,
+                                                   selector: #selector(SiteCreationRotatingMessageView.displayNextMessage),
+                                                   userInfo: nil,
+                                                   repeats: true)
+    }
+
+
+    /// Cancels the timer, and stops the rotating animations
+    func stopAnimating() {
+        guard let timer = animationTimer else {
+            return
+        }
+
+        timer.invalidate()
+        animationTimer = nil
+
+        reset()
+    }
+
+    // MARK: - Internal
+    /// Resets the visible index to 0, and changes the visible message to the first one
+    internal func reset() {
+        visibleIndex = 0
+
+        //Update the status label with the first status message in the list
+        statusLabel.text = statusMessages[0]
+    }
+
+    /// Updates the status label text with the next message to be displayed
+    /// then updates the current visible index
+    internal func updateStatusLabelWithNextMessage() {
+        let nextIndex = self.nextVisibleIndex
+        let statusMessage = self.statusMessages[nextIndex]
+
+        self.updateStatus(message: statusMessage)
+
+        self.visibleIndex = nextIndex
+    }
+
+    /// Updates the status label/accessiblity label with the provided text
+    /// - Parameter message: The text to be displayed
+    internal func updateStatus(message: String) {
+        let statusMessage = NSLocalizedString(message,
+                                              comment: "User-facing string, presented to reflect that site assembly is underway.")
+
+        self.statusLabel.text = statusMessage
+        self.statusLabel.accessibilityLabel = statusMessage
```

Comment:
Since this is a plain `UIlabel`, the `accessibilityLabel` will have the text content. I guess we can remove this line.

Reply:
Okay! I just mimicked it from the `SiteAssemblyContentView`